src/main/java/com/disi/TravelPoints/service/LandmarkService.java

```diff
@@ -51,6 +54,26 @@ public void add(AddLandmarkRequest request){
         landmarkRepository.save(landmark);
     }
 
```

Comment:
I think we could use the @Transactional annotation to ensure that the data in the database is always consistent.

Reply:
I think this does not apply only on this service. @MatiocBogdanDarius @RazvanMadar you might want to check this out

Reply:
The annotation should appear where changes are made to an entity

Reply:
Why? Maybe only when are 2 operations on DB like here (retrieve + save)?

Reply:
(For this method) I don't think it is necessary to add the annotation `@Transactional` because in this method only one method is called that changes the state of the database (`save` - which is already a transactional method [see `SimpleJpaRepository`]) . I think that the presence of this annotation is justified only when several methods are called to modify the state of the database or the same method is called several times.

Reply:
I agree with this. But, as far as I know, as a good practice, especially for update operations, it is good to put such an annotation to make sure that either all the changes you want to make are made successfully, or not at all

Reply:
Yes, but as @MatiocBogdanDarius said, **save** method is already transactional, so it's a bit unnecessary

Reply:
In the present case, there are no problems with partial executions, as I said before, the `save` method is transactional. The problem that could arise would be between getting the information and updating it. If we had operations that increase/decrement certain values, the lack of annotation could be a problem. In the present case, I don't think it is absolutely necessary to add the annotation. Let's not forget that using the `@Transactional' annotation has a negative effect on performance.